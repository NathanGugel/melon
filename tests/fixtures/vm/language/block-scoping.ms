let c = "top-level c";
let b = "top-level b";

if (true) {
    let b = "test";
    io.print(b);

    {
        let b = "nested b";
        io.print(b);

        let c = "nested c";
        io.print(c);

        let inner = "inner only";
        io.print(inner);
    }

    io.print(c);
    io.print(inner);

    b = "test changed";
    io.print(b);

    let sameSlots = "value";
    io.print(sameSlots);
}

io.print(b);

{
    func myFunc => {
        io.print("This is my func");
    }

    {
        func myFunc |x| => {
            io.print("This is my inner func: " .. string.toString(x));
        }

        myFunc(10);
    }

    myFunc();

    func myOtherFunc =>  {
        io.print("My other func");
    }

    myOtherFunc();
}

for (let i in 1..5) {
    io.print("outer: " .. string.toString(i));

    for (let i in 1..5) {
        io.print("inner: " .. string.toString(i));
    }
}

for (let i in 5..0) {
    io.print(string.toString(i));
}

{
    let a = "outer a";
    let b = "outer b";

    func myFunc => {
        let b = "inner b";
        io.print(b);
        
        {
            let b = "inner inner b";
            io.print(b);
        }

        io.print(a);
        io.print(c);
    }

    myFunc();
}
