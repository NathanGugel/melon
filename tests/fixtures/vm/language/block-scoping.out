test
nested b
nested c
inner only
top-level c
null
test changed
value
top-level b
This is my inner func: 10
This is my func
My other func
outer: 1
inner: 1
inner: 2
inner: 3
inner: 4
outer: 2
inner: 1
inner: 2
inner: 3
inner: 4
outer: 3
inner: 1
inner: 2
inner: 3
inner: 4
outer: 4
inner: 1
inner: 2
inner: 3
inner: 4
4
3
2
1
0
inner b
inner inner b
outer a
top-level c
